query-translator/tests/translate.rs

```diff
@@ -1091,28 +1091,35 @@ fn test_project_aggregates() {
     let SQLQuery { sql, args } = translate(&schema, query);
 
     // No outer DISTINCT: we aggregate or group by every variable.
-    assert_eq!(sql, "SELECT `?e` AS `?e`, max(`?t`) AS `(max ?t)` \
+    assert_eq!(sql, "SELECT * \
                      FROM \
-                     (SELECT DISTINCT \
-                      `datoms00`.e AS `?e`, \
-                      `datoms00`.v AS `?t` \
-                      FROM `datoms` AS `datoms00` \
-                      WHERE `datoms00`.a = 99) \
-                      GROUP BY `?e`");
+                     (SELECT `?e` AS `?e`, max(`?t`) AS `(max ?t)` \
+                      FROM \
+                      (SELECT DISTINCT \
+                       `datoms00`.e AS `?e`, \
+                       `datoms00`.v AS `?t` \
+                       FROM `datoms` AS `datoms00` \
+                       WHERE `datoms00`.a = 99) \
+                      GROUP BY `?e`) \
+                     WHERE `(max ?t)` IS NOT NULL");
     assert_eq!(args, vec![]);
 
     let query = r#"[:find (max ?t)
                     :with ?e
                     :where
                     [?e :foo/bar ?t]]"#;
     let SQLQuery { sql, args } = translate(&schema, query);
-    assert_eq!(sql, "SELECT max(`?t`) AS `(max ?t)` \
+    assert_eq!(sql, "SELECT * \
                      FROM \
-                     (SELECT DISTINCT \
-                      `datoms00`.v AS `?t`, \
-                      `datoms00`.e AS `?e` \
-                      FROM `datoms` AS `datoms00` \
-                      WHERE `datoms00`.a = 99)");
+                     (SELECT max(`?t`) AS `(max ?t)` \
+                      FROM \
+                      (SELECT DISTINCT \
+                       `datoms00`.v AS `?t`, \
+                       `datoms00`.e AS `?e` \
+                       FROM `datoms` AS `datoms00` \
+                       WHERE `datoms00`.a = 99)\
+                      ) \
+                     WHERE `(max ?t)` IS NOT NULL");
```

Comment:
I’d like two more tests:

- that we don’t null check `count`. That test will fail, then you must decide what to change!
- that we don’t null-check `the` or grouped vars. Same.

Reply:
> that we don’t null check count. That test will fail, then you must decide what to change!

I thought about this, and the null check is harmless in general.  A tiny bit of SQLite overhead.  

Doing better is irritating: we're fixing this up _way_ after the point we should be, due to us handling aggregates very early and then dropping all the information.  I thought of renaming `::Expression` to be more particular to aggregates, and then splitting that case based on the type of aggregate... but I really don't think this is worth it.

`the` and grouped vars would change other tests, and they don't produce `::Expression` enums, so this won't fail.  I'm happy to add a test specifically here to show that it doesn't happen.

I'll give the split a stab and if I like it I'll land it; if not, this'll get unblock me.